compiler/qsc/src/codegen/tests.rs

```diff
@@ -744,6 +883,151 @@ mod adaptive_ri_profile {
         "#]].assert_eq(&qir);
     }
 
+    #[test]
+    fn custom_measurement_generates_correct_qir() {
```

Comment:
It seems like there are two tests that have the same name, I'm surprised the compiler doesn't produce errors in this case. Additionally, it looks like the only difference between this test and the other one with the same name is tha capabilities. Is there something specific we want to test regarding behavior that would be affected by other flags that are not `TargetCapabilityFlags::Adaptive`?

Reply:
There were other tests with repeated names in that file. The reason is that the name of the test is its full path:
 ```rust
 adaptive_ri_profile::custom_measurement_generates_correct_qir
 ```
 
 which differentiates it from:
 ```rust
 base::custom_measurement_generates_correct_qir
 ```